src/util/NuulsUploader.cpp

```diff
@@ -0,0 +1,205 @@
+{
+    uploadImageToNuuls(imageData.data, channel, textEdit, imageData.type);
+}
+void uploadImageToNuuls(QByteArray imageData, ChannelPtr channel,
+                        ResizingTextEdit &textEdit, std::string format)
+{
+    QByteArray dataToSend;
+    const char *boundary = "thisistheboudaryasd";
+    static QUrl url(Env::get().imagePasteSiteUrl);
+
+    dataToSend.insert(0, "--");
+    dataToSend.append(boundary);
+    std::string temp = "\r\n"
+                       "Content-Disposition: form-data; name=\"attachment\"; "
+                       "filename=\"control_v.%\"\r\n"
+                       "Content-Type: image/%\r\n"
+                       "\r\n";
+    boost::replace_all(temp, "%", format);
+
+    dataToSend.append(temp.c_str());
+    dataToSend.append(imageData);
+    dataToSend.append("\r\n--");
+    dataToSend.append(boundary);
+    dataToSend.append("\r\n");
+
+    NetworkRequest(url, NetworkRequestType::Post)
+        .header("Content-Type", (std::string("multipart/form-data; boundary=") +
+                                 std::string(boundary))
+                                    .c_str())
+
+        .payload(dataToSend)
+        .onSuccess([&textEdit, channel](NetworkResult result) -> Outcome {
+            textEdit.insertPlainText(result.getData() + QString(" "));
+            //            this->input_->ui_.textEdit->insertPlainText(result.getData());
+            if (uploadQueue.size())
+            {
+                channel->addMessage(makeSystemMessage(
+                    QString("Your image has been uploaded. %1 left. Please "
+                            "wait until all of them are uploaded. About %2 "
+                            "seconds left.")
+                        .arg(uploadQueue.size())
+                        .arg(uploadQueue.size() * 3)));
+                // Argument number 2 is the ETA.
+                // 2 seconds for the timer that's there not to spam Nuuls' server
+                // and 1 second of actual uploading.
+            }
+            else
+            {
+                channel->addMessage(makeSystemMessage(
+                    QString("Your image has been uploaded.")));
+            }
+            isUploading = false;
+            QTimer::singleShot(2000, [channel, &textEdit]() {
+                if (uploadQueue.size())
+                {
+                    uploadImageToNuuls(uploadQueue.front(), channel, textEdit);
+                    uploadQueue.pop();
+                }
+            });
+            return Success;
+        })
+        .onError([channel](int error_code) -> bool {
+            channel->addMessage(makeSystemMessage(
+                QString("An error happened while uploading your image: %1")
+                    .arg(error_code)));
+            isUploading = false;
+            return true;
+        })
+        .execute();
+}
+QString getImageFileFormat(QString path)
+{
+    static QStringList LIST_OF_IMAGE_FORMATS = {".png", ".jpg", ".jpeg"};
```

Comment:
`LIST_OF_IMAGE_FORMATS ` should be in camelCase. Only `#defines` should be UPPER_SNAKE_CASE.